tests/manage/test_expand_cluster_by_adding_backend_vol_rbd.py

```diff
@@ -0,0 +1,378 @@
+
+    Returns:
+        True, if the expected_osd_pod_count matches with the actual_osd_count observed.
+    """
+    actual_pods_added = 0
+
+    for pod_after in after:
+        found = False
+        for pod_before in before:
+            logging.info(f"pod after = {pod_after.name}, pod before = {pod_before.name}")
+            if pod_after.name in pod_before.name:
+                found = True
+                break
+        if not found:  # we found the newly added osd pod
+            pod_status = pod_after.get()
+            if (pod_status['status']['containerStatuses'][0]['restartCount'] == 0 and
+                    pod_status['status']['phase'] == constants.STATUS_RUNNING):
+                logging.info(f"Found the newly added pod {pod_after.name} and is running successfully")
+                actual_pods_added += 1
+    if expected_osd_pod_count == actual_pods_added:
+        return True
+    else:
+        logging.error(f"Expected to find {expected_osd_pod_count} osd pods but found {actual_pods_added}")
+        return False
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    self = request.node.cls
+
+    def finalizer():
+        tear_down(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def tear_down(self):
+
+    # Delete app pods
+    for p in self.pods:
+        assert p.delete()
+
+    # Delete the PVCs
+    assert pvc.delete_pvcs(self.pvcs)
+
+    # TBD: Delete app-pods namespace. Waiting for the patch
+
+    # Delete storage class
+    helpers.delete_all_storageclass()
+
+    # Delete secret
+    self.secret.delete()
+
+    # TBD: To remove the backend disk that was added as part of this test and its corresponding osd pod.
+    #       NOTE: We do not have the rook way of removing the osd pod after its backend disk is removed.
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+class TestExpandClusterByAddingBackendVol(ManageTest):
+    @pytest.mark.polarion_id("OCS-356")
+    def test_expand_cluster_by_adding_backend_vol_rbd(self):
+        """
+        This script automates the following test case:
+        https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-356
+        """
+        number_of_osds_to_add = 2  # TBD: change this to parameterized format later. Accept it from cmdline
+
+        # Get the osd pod names, osd count and restart count for all pods before adding new osd
+        osd_pods_before = pod.get_osd_pods()
+        original_osd_count = len(osd_pods_before)
+        pod_restart_count_before = get_pod_restarts_count()
+
+        # Run fio on the pods created
+        for p in self.pods:
+            # Run fio with data integrity check enabled for 600 seconds.
+            p.run_io('fs', '50M', 'wv', 75, 1, 900)
+
+        logger.info("Allowing IOs to continue for few minutes. Sleeping....")
+        sleep(60)  # Allow IOs to soak for at least 1-2 minutes before we add new disk.
```

Comment:
I don't like sleep so can you explain what are we waiting for?
To have some data wrriten in the PV?

* If we know what we are waiting for, we can create a function to sample that instead of just sleeping (which is waising time)